src/taxonomy/import-tags/data/utils.test.js

```diff
@@ -0,0 +1,301 @@
+      return promise;
+    });
+  });
+
+  describe('import tags', () => {
+    it('should call the api and show success alert', async () => {
+      jest.spyOn(window, 'confirm').mockReturnValueOnce(true);
+      jest.spyOn(window, 'alert').mockImplementation(() => {});
+
+      const promise = importTaxonomyTags(1, intl).then(() => {
+        expect(importTags).toHaveBeenCalledWith(1, 'mockFile');
+        expect(window.alert).toHaveBeenCalledWith('Taxonomy imported successfully');
+      });
+
+      // Capture the onChange handler from the file input element
+      const onChange = mockAddEventListener.mock.calls[0][1];
+      const mockTarget = {
+        target: {
+          files: [
+            'mockFile',
+          ],
+        },
+      };
+
+      onChange(mockTarget);
+
+      return promise;
+    });
+
+    it('should abort the call to the api without file', async () => {
+      const promise = importTaxonomyTags(1, intl).then(() => {
+        expect(importTags).not.toHaveBeenCalled();
+      });
+
+      // Capture the onChange handler from the file input element
+      const onChange = mockAddEventListener.mock.calls[0][1];
+      const mockTarget = {
+        target: {
+          files: [null],
+        },
+      };
+
+      onChange(mockTarget);
+      return promise;
+    });
+
+    it('should abort the call to the api if file closed', async () => {
+      const promise = importTaxonomyTags(1, intl).then(() => {
+        expect(importTags).not.toHaveBeenCalled();
+      });
+
+      // Capture the onCancel handler from the file input element
+      const onCancel = mockAddEventListener.mock.calls[1][1];
+
+      onCancel();
+      return promise;
+    });
+
+    it('should abort the call to the api when cancel the confirm dialog', async () => {
+      jest.spyOn(window, 'confirm').mockReturnValueOnce(null);
+
+      const promise = importTaxonomyTags(1, intl).then(() => {
+        expect(importTags).not.toHaveBeenCalled();
+      });
+
+      // Capture the onChange handler from the file input element
+      const onChange = mockAddEventListener.mock.calls[0][1];
+      const mockTarget = {
+        target: {
+          files: [
+            'mockFile',
+          ],
+        },
+      };
+
+      onChange(mockTarget);
+
+      return promise;
+    });
+
+    it('should call the api and return error alert', async () => {
+      jest.spyOn(window, 'confirm').mockReturnValueOnce(true);
+      importTags.mockRejectedValue(new Error('test error'));
+
+      const promise = importTaxonomyTags(1, intl).then(() => {
+        expect(importTags).toHaveBeenCalledWith(1, 'mockFile');
+      });
+
+      // Capture the onChange handler from the file input element
+      const onChange = mockAddEventListener.mock.calls[0][1];
+      const mockTarget = {
+        target: {
+          files: [
+            'mockFile',
+          ],
+        },
+      };
+
+      onChange(mockTarget);
+
+      return promise;
+    });
+  });
+});
```

Comment:
Additionally, I think some of the code here is quite repetitive. Here are a few suggestions:

- I see repetition of `const onChange ...` this could be moved to a utility function like: `mockOnChange` that cane get take the files array as a parameter. 
- OR. Even better, just mock the `selectFile` function with a known return value. You can test the `selectFile` function separately. This should also clean up and simplify a lot of this code.

Reply:
I expect that this whole file will be replaced. The `importTaxonomyTags` was already replaced in https://github.com/open-craft/frontend-app-course-authoring/pull/19 ([here](https://github.com/open-craft/frontend-app-course-authoring/pull/19/files#r1428382384)). So, I went for the `/* istanbul ignore next */ ` route to save time. Let me know if you are ok with this!

Reply:
Sure, that makes sense!